test/middleware/webhook-timeout.test.js

```diff
@@ -0,0 +1,32 @@
+const webhookTimeout = require('../../lib/middleware/webhook-timeout')
+
+describe('Webhook Timeout', () => {
+  it('sets timedout context with seconds', async () => {
+    const context = {}
+    await webhookTimeout(async () => { await sleep(3) }, 1)(context)
```

Comment:
Just a reminder that `sleep(3)` means sleep 3 milliseconds. I'm not a huge fan of sleeping within tests, but sleeping milliseconds feels alright.

Reply:
I thought it was sleep 3 seconds for a moment 😆 phew! yea, that seems fine to me!

Reply:
yeah I used `ms` as the argument to sleep, but I thought about changing it to `sleepMs` instead

Reply:
Ah I actually like that better and that's a little more Node-y too. Thanks!

Reply:
I talked with @jules2689 in slack. I think this is a good suggestion. But I'm struggling with how to change up the tests. I definitely welcome any help here.

There's effectively two callbacks I care about here:

1. That the timedout is set within the setTimeout
2. That the timer is cleared in both a succesful and error case

I think at least the first one needs to be some sort of async function that we wait on?

I may also still need to use `sleep` at least for the timed out scenario (since that's being called within the closure being wrapped).

I'll mull on it overnight and see if any inspiration comes to me.

❤️ for the suggestion @jules2689